docs/docs/guides/developer_guides/smart_contracts/profiling_transactions.md

```diff
@@ -0,0 +1,79 @@
+---
+title: Profiling Transactions
+sidebar_position: 5
+tags: [contracts, profiling]
+---
+
+# Profiling Transactions
+
+An Aztec transaction typically consists of a private and a public part. The private part is where the user executes contract logic within the PXE and generates a proof of execution, which is then sent to the sequencer.
+
+Since proof generation is an expensive operation that needs to be done on the client side, it is important to optimize the private contract logic. That is, it is desirable to keep the gate count of circuits representing the private contract logic as low as possible.
+
+A private transaction can involve multiple function calls. It starts with an account `entrypoint()` which may call several private functions to execute the application logic, which in turn might call other functions. Moreover, every private function call has to go through a round of kernel circuits. Read more about the transaction lifecycle [here](../../../aztec/concepts/transactions.md).
+
+In this guide, we will look at how to profile the private execution of a transaction, allowing you to get the gate count of each private function within the transaction, including the kernel circuits.
+
+## Prerequisites
+
+- `aztec-nargo` installed (go to [Sandbox section](../../../reference/developer_references/sandbox_reference/sandbox-reference.md) for installation instructions)
+- `aztec-wallet` installed (installed as part of the Sandbox)
+- Aztec Sandbox running with **proving enabled** (go to [Sandbox PXE Proving](../local_env/sandbox_proving.md) for instructions)
```

Comment:
This is a requirement? Bummer, this can take a long time to set up and run, esp. if we are deploying additional accounts as described below.

Any ideas for how we can make profiling faster?

Reply:
Yea, this is not very good now, but proving needs to be enabled at the moment for this to work. 

Proving is not really needed to get the gate count technically, but our code is currently structured in a way that you get the real BB prover (which will provide the gate count) only if proving is enabled - otherwise it defaults to a mock prover.
I will try to address this separately.